File: back/src/main/java/me/yourcaryourway/YourCarYourWay/models/Message.java
package me.yourcaryourway.YourCarYourWay.models;

import jakarta.annotation.Nullable;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.*;
import org.apache.commons.lang3.builder.ToStringExclude;

import java.sql.Timestamp;

@Entity
@Table(name = "MESSAGE")
@Data
@EqualsAndHashCode(of = {"id"}, callSuper = false)
@Builder
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Message {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @ToStringExclude
    private Long id;

    @Nullable
    private String subject;

    @NotNull
    private String text;

    @ManyToOne
    @JoinColumn(name = "type_id", nullable = false)
    @NotNull
    private MessageType type;

    @Column(name = "is_read")
    @NotNull
    private Boolean isRead;

    @ManyToOne
    @JoinColumn(name = "sender_id", nullable = false)
    @NotNull
    private User sender;

    @ManyToOne
    @JoinColumn(name = "receiver_id", nullable = false)
    @NotNull
    private User receiver;

    @OneToOne
    @JoinColumn(name = "message_parent_id")
    @Nullable
    private Message parent;

    @OneToOne(mappedBy = "parent")
    private Message child;

    @Column(name = "created_at")
    @NotNull
    private Timestamp createdAt;
}

File: front/src/app/models/Message.ts
export class Message {
    private id : number;
    private parentId : number;
    private content : string;
    private isRead : boolean;
    private sender : string;
    private receiver : string;
    private createdAt : string;

	constructor($id: number, $parentId: number, $content: string, $isRead: boolean, $sender: string, $receiver: string, $createdAt: string) {
		this.id = $id;
		this.parentId = $parentId;
		this.content = $content;
		this.isRead = $isRead;
		this.sender = $sender;
		this.receiver = $receiver;
		this.createdAt = $createdAt;
	}

    /**
     * Getter $id
     * @return {number}
     */
	public get $id(): number {
		return this.id;
	}

    /**
     * Getter $parentId
     * @return {number}
     */
	public get $parentId(): number {
		return this.parentId;
	}

    /**
     * Getter $content
     * @return {string}
     */
	public get $content(): string {
		return this.content;
	}

    /**
     * Getter $isRead
     * @return {boolean}
     */
	public get $isRead(): boolean {
		return this.isRead;
	}

    /**
     * Getter $sender
     * @return {string}
     */
	public get $sender(): string {
		return this.sender;
	}

    /**
     * Getter $receiver
     * @return {string}
     */
	public get $receiver(): string {
		return this.receiver;
	}

    /**
     * Getter $createdAt
     * @return {string}
     */
	public get $createdAt(): string {
		return this.createdAt;
	}

    /**
     * Setter $id
     * @param {number} value
     */
	public set $id(value: number) {
		this.id = value;
	}

    /**
     * Setter $parentId
     * @param {number} value
     */
	public set $parentId(value: number) {
		this.parentId = value;
	}

    /**
     * Setter $content
     * @param {string} value
     */
	public set $content(value: string) {
		this.content = value;
	}

    /**
     * Setter $isRead
     * @param {boolean} value
     */
	public set $isRead(value: boolean) {
		this.isRead = value;
	}

    /**
     * Setter $sender
     * @param {string} value
     */
	public set $sender(value: string) {
		this.sender = value;
	}

    /**
     * Setter $receiver
     * @param {string} value
     */
	public set $receiver(value: string) {
		this.receiver = value;
	}

    /**
     * Setter $createdAt
     * @param {string} value
     */
	public set $createdAt(value: string) {
		this.createdAt = value;
	}

}